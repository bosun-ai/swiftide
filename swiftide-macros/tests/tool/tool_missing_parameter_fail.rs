#[swiftide_macros::tool(
    description = "My first tool",
    param(name = "Message", description = "A message for testing")
)]
async fn basic_tool(_agent_context: &dyn AgentContext, msg: &str) -> Result<ToolOutput, ToolError> {
    Ok(format!("Hello {msg}").into())
}

fn main() {}
