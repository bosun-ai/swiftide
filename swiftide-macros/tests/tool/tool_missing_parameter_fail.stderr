error: Arguments in spec and in function do not match:
        The following parameters are missing from the function signature: ["Message"], The following parameters are missing from the spec: ["msg"]
 --> tests/tool/tool_missing_parameter_fail.rs:1:1
  |
1 | / #[swiftide_macros::tool(
2 | |     description = "My first tool",
3 | |     param(name = "Message", description = "A message for testing")
4 | | )]
  | |__^
  |
  = note: this error originates in the attribute macro `swiftide_macros::tool` (in Nightly builds, run with -Z macro-backtrace for more info)
